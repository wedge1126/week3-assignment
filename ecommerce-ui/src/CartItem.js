import React from 'react';
import PropTypes from 'prop-types';

export default class CartItem extends React.Component {
  render() {
    const { home, onRemoveItem, idx } = this.props;
    const currencyFormat = new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD',
        minimumFractionDigits: 0,
        maximumFractionDigits: 0
    });

    return <div className="cart-item">
        <div className="cart-item-body">
            <div className="cart-item-type">
                <span>{home.houseType}</span> &#8226; <span>{home.location.city}, {home.location.country}</span>
            </div>
            <div className="cart-item-cost">
                {currencyFormat.format(home.payment.cost)}/night
            </div>
            <div>
                <span className="cart-item-title">{home.title}</span>
                <button className="cart-item-button" onClick={() => onRemoveItem(idx)} >Remove</button>
            </div>
        </div>
    </div>
  }
}

CartItem.propTypes = {
    home: PropTypes.shape({
        title: PropTypes.string.isRequired,
        houseType: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
        location: PropTypes.shape({
            city: PropTypes.string.isRequired,
            country: PropTypes.string.isRequired
        }).isRequired,
        payment: PropTypes.shape({
            cost: PropTypes.number.isRequired,
            description: PropTypes.string.isRequired
        }).isRequired,
        host: PropTypes.shape({
            name: PropTypes.string.isRequired,
            isSuperhost: PropTypes.bool.isRequired
        }).isRequired,
        rating: PropTypes.shape({
            stars: PropTypes.number.isRequired,
            reviews: PropTypes.number.isRequired
        })
    }).isRequired,
    onRemoveItem: PropTypes.func.isRequired,
    idx: PropTypes.number.isRequired
}