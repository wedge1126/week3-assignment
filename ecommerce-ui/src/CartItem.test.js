import React from 'react';
import { shallow, mount } from 'enzyme';

import CartItem from './CartItem';

const testHome = {
    "title": "Quaint 31,400 kmÂ³ home above lovely forest",
    "houseType": "Entire Space Station",
    "image": "http://www.islandbreath.org/2015Year/02/150205deathstar.jpg",
    "location": {
        "city": "Endor",
        "country": "Endor System"
    },
    "payment": {
        "cost": 3200000000,
        "description": ""
    },
    "host": {
        "name": "Sheev",
        "isSuperhost": true
    },
    "rating": {
        "stars": 3.2,
        "reviews": 501
    }
};

describe('CartItem', () => {
    it('should render the test data correctly', () => {
        const onRemoveItem = jest.fn();
        const idx = 1;
        const component = shallow(<CartItem home={testHome} onRemoveItem={onRemoveItem} idx={idx} />);
        
        expect(component).toMatchSnapshot();
    });

    it('should call onAddToCart when button is clicked', () => {
        const onRemoveItem = jest.fn();
        const idx = 1;
        const component = mount(<CartItem home={testHome} onRemoveItem={onRemoveItem} idx={idx} />);
        
        component.find('button.cart-item-button').simulate('click');

        expect(onRemoveItem).toHaveBeenCalled();

        component.unmount();
    });
});