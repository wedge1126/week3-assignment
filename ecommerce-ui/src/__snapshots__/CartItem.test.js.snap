// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render the test data correctly 1`] = `
<div
  className="cart-item"
>
  <div
    className="cart-item-body"
  >
    <div
      className="cart-item-type"
    >
      <span>
        Entire Space Station
      </span>
       • 
      <span>
        Endor
        , 
        Endor System
      </span>
    </div>
    <div
      className="cart-item-cost"
    >
      $3,200,000,000
      /night
    </div>
    <div>
      <span
        className="cart-item-title"
      >
        Quaint 31,400 km³ home above lovely forest
      </span>
      <button
        className="cart-item-button"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
